example/contrib/socks-client/socks/impl/async_handshake_v5.hpp

```diff
@@ -397,6 +404,35 @@ class socks5_op : public base_type, public net::coroutine
     int step_ = 0;
 };
 
+struct async_handshake_v5_initiator
```

Comment:
I think in the latest iteration, the initiating function object has a nested `executor_type` which enables some special thing.